ASD/graph/topological_sorting.py

```
@@ -0,0 +1,48 @@
+import sys
+
+class Node:
+    def __init__(self, value):
+        self.value = value
+        self.color = "white"
+
+class AdjacencyList:
+    def __init__(self, V):  # O(V)
+        self.graph = [[] for _ in range(V)]
+    
+    def add(self, v, u):    # O(1)
+        self.graph[v.value].append(u)
+    
+    def neighbors(self, node): # O(1)
+        return self.graph[node.value]
+    
+def topological_sorting(G, V):
+    L = []
+    for u in V:
+        if u.color == "white":
+            DFS(G, L, u)
+    L.reverse()
+    return L
+
+def DFS(G, L, u):
+    u.color = "grey"
+    for v in G.neighbors(u):
+        if v.color == "grey":   # CYCLE
+            print("Graph is not a DAG")
+            sys.exit()
```

you should raise an exception here